src/lib/storage/constraints/table_unique_constraint.cpp

```diff
@@ -0,0 +1,14 @@
+#include "table_unique_constraint.hpp"
+
+namespace opossum {
+
+TableUniqueConstraint::TableUniqueConstraint() : TableKeyConstraint({}, KeyConstraintType::NONE) {}
```

Comment:
Why is "none" a valid constraint type?

Reply:
Simple declarations like this
```cpp
  TableUniqueConstraint _unique_constraint_a;
  TableUniqueConstraint _unique_constraint_b_c;
  TableUniqueConstraint _unique_constraint_x;
  TableUniqueConstraint _unique_constraint_y;
```
require me to define a default constructor. Full initialization can not take place on startup because the SetUp() routine has to create tables first.

I did not want to create a dummy PRIMARY KEY or UNIQUE constraint. Therefore, I added NONE as another type. How would you do it?

Reply:
Tests should usually not impact the design of the main code. If it's only those four lines, maybe we can just use optional<>?

Reply:
Good suggestion! I'll go with optional<>.